### PR TITLE
Build and upload LGPU wheels for Linux aarch64

### PR DESCRIPTION
**Pull request by maliasadi**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
Add `aarch64` wheels recipes for `"lightning.gpu"`.

**Description of the Change:**

**Benefits:**
Users can pip install LGPU on ARM boxes.

**Possible Drawbacks:**

**Related GitHub Issues:**
[sc-60048]
[sc-60750]

### CODE REVIEW

File: .github/workflows/wheel_lgpu_arm.yml [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+          # MacOS specific build settings
+          CIBW_BEFORE_ALL_MACOS: |
+            brew uninstall --force oclint
+
+          # Python build settings
+          CIBW_BEFORE_BUILD: |
+            python -m pip install pybind11 ninja cmake~=3.24.0 setuptools scipy
+
+          CIBW_ENVIRONMENT: |
+            CMAKE_ARGS="-DCMAKE_CXX_COMPILER_TARGET=arm64-apple-macos11 -DCMAKE_SYSTEM_NAME=Darwin -DCMAKE_SYSTEM_PROCESSOR=ARM64 -DENABLE_OPENMP=OFF" \
```
Comment (multiphaseCFD) [inline .github/workflows/wheel_lgpu_arm.yml:94]: hmmm, could you please explain why the `CMAKE_CXX_COMPILER_TARGET` flag is set as `arm64-apple-macos11`?  macos-11 doesn't support nvidia gpus, right?

File: .github/workflows/wheel_lgpu_arm.yml [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    needs: [mac-set-matrix-arm]
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-12]
+        arch: [arm64]
+        pl_backend: ["lightning_gpu"]
+        cibw_build: ${{fromJson(needs.mac-set-matrix-arm.outputs.python_version)}}
+    timeout-minutes: 30
+    name: macos-latest::arm64 - ${{ matrix.pl_backend }} (Python ${{ fromJson('{ "cp39-*":"3.9","cp310-*":"3.10","cp311-*":"3.11","cp312-*":"3.12" }')[matrix.cibw_build] }})
```
Comment (multiphaseCFD) [inline .github/workflows/wheel_lgpu_arm.yml:66]: Not sure why do we want to build it with `macos-latest::arm64`. Could you please elaborate more about it?

**Review by multiphaseCFD — COMMENTED** [review]
Thanks @maliasadi ! Just leave some comments for the first round.

File: .github/workflows/wheel_lgpu_arm.yml [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+            CMAKE_ARGS="-DCMAKE_CXX_COMPILER_TARGET=arm64-apple-macos11 -DCMAKE_SYSTEM_NAME=Darwin -DCMAKE_SYSTEM_PROCESSOR=ARM64 -DENABLE_OPENMP=OFF" \
+            PL_BACKEND="${{ matrix.pl_backend }}"
+
+          CIBW_BEFORE_TEST: |
+            python -m pip install -r requirements-tests.txt
+            SKIP_COMPILATION=True PL_BACKEND="lightning_qubit" python -m pip install . -vv
+
+          CIBW_TEST_COMMAND: |
+            DEVICENAME=`echo ${{ matrix.pl_backend }} | sed "s/_/./g"`
+            pl-device-test --device=${DEVICENAME} --skip-ops -x --tb=short --no-flaky-report
```
Comment (LuisAlfredoNu) [inline .github/workflows/wheel_lgpu_arm.yml:103]: Do we have a test for this hardware configuration? 
How we are sure that the code will use the GPU of Mx chips?

Comment (maliasadi) [reply .github/workflows/wheel_lgpu_arm.yml:103]: @LuisAlfredoNu Not yet as it requires more tests locally.. I'll tag you and others when it's ready to review :)

**Review by multiphaseCFD — APPROVED** [review]
Thanks @maliasadi ! LGTM! It's a  really nice feature to have

**Review by mlxd — APPROVED** [review]
All good --- check on the following:
* wget/openssh needed?
* Update auditwheel libs?
* Update qemu
* The remaining todos

**Pull request merged**